Inferences/Induction.cpp

```diff
@@ -201,12 +202,38 @@ TermList TermReplacement::transformSubterm(TermList trm)
 {
   CALL("TermReplacement::transformSubterm");
 
-  if(trm.isTerm() && trm.term()==_o){
-    return _r;
+  if(trm.isTerm()) {
+    auto t = trm.term();
+    if (t==_o){
+      return _r;
+    }
+    if (_replaceSkolems) {
+      unsigned f = t->functor();
+      if (env.signature->getFunction(f)->skolem()) {
+        unsigned v;
+        if (!_tv.find(t,v)) {
+          v = (*_v)++;
+          _tv.insert(t,v);
+        }
+        return TermList(v,false);
+      }
+    }
   }
   return trm;
 }
 
+VList* TermReplacement::getUsedVars() const
```

Comment:
No, I haven't tried sorting it. As far as I can see, they go through NNF unchanged and then the order of introduced Skolems will be dependent on the order of variables in NewCNF. Do you think it matters? Maybe it can block Skolem reuse?

Reply:
I see... Well, since `Induction` has its own `notDone` and `notDoneInt` methods to determine if a certain induction formula was added and if so, it doesn't generate anything, Skolem reuse does not play a role here yet (except when induction formulas are added as input?). That we cannot use an induction formula twice is by the way the source of losing (not many but still) some proofs.

If we want to solve this issue later, ideally, Skolem reuse would be enough to avoid generating duplicate clauses but then still the formulas would exist and use up memory if I understand correctly. A solution I've been working on skips the clausification part and uses the generated clauses right away from an index. The question is whether we should go for a more general approach which comes with additional issues like this, having to sort the list of variables, or solve this issue inside `Induction`.

Reply:
OK..."future work" and we can pass this to Martin then?